utils/httpx/http_test.go

```diff
@@ -27,6 +28,38 @@ func TestNewTrace(t *testing.T) {
 	assert.Equal(t, time.Date(2019, 10, 7, 15, 21, 31, 123456789, time.UTC), trace.EndTime)
 }
 
+func TestDisallowedHosts(t *testing.T) {
+	defer httpx.SetRequestor(httpx.DefaultRequestor)
+
+	disallowedHosts := []string{"localhost", "127.0.0.1", "::1"}
+
+	httpx.SetRequestor(httpx.NewMockRequestor(map[string][]httpx.MockResponse{
+		"https://temba.io": []httpx.MockResponse{
+			httpx.NewMockResponse(200, nil, ``, 1),
+		},
+	}))
+
+	call := func(url string) (*httpx.Trace, error) {
+		request, _ := http.NewRequest("GET", url, nil)
+		return httpx.DoTrace(http.DefaultClient, request, nil, disallowedHosts, -1)
+	}
+
+	_, err := call("https://temba.io")
+	assert.NoError(t, err)
+
+	_, err = call("https://localhost/path")
+	assert.EqualError(t, err, "requests to host localhost are disallowed")
+
+	_, err = call("https://LOCALHOST:80")
+	assert.EqualError(t, err, "requests to host localhost are disallowed")
+
+	_, err = call("https://127.0.0.1")
+	assert.EqualError(t, err, "requests to host 127.0.0.1 are disallowed")
+
+	_, err = call("https://[::1]:80")
```

Comment:
I need to read up on IPv6 addresses but I assume that we need to do some normalization as same addresses can have slightly different representations